src/Microsoft.Diagnostics.Monitoring.WebApi/Controllers/DiagController.cs

```diff
@@ -592,28 +592,75 @@ public ActionResult<DotnetMonitorInfo> GetInfo()
 
             return await InvokeForProcess(async processInfo =>
             {
-                IInProcessOperation operation = _captureParametersFactory.Create(processInfo.EndpointInfo, configuration, duration);
-                return await InProcessResult(Utilities.ArtifactType_Parameters, processInfo, operation, tags);
-            }, processKey, Utilities.ArtifactType_Parameters);
+                return await InProcessResult(
+                    Utilities.ArtifactType_RequestParameters,
+                    processInfo,
+                    requestId => _captureParametersFactory.Create(requestId, processInfo.EndpointInfo, configuration, duration),
+                    tags);
+            }, processKey, Utilities.ArtifactType_RequestParameters);
         }
 
-        [HttpGet("stacks", Name = nameof(CaptureStacks))]
-        [ProducesWithProblemDetails(ContentTypes.ApplicationJson, ContentTypes.TextPlain, ContentTypes.ApplicationSpeedscopeJson)]
+        [HttpGet("parameters", Name = nameof(GetCapturedParameters))]
+        [ProducesWithProblemDetails(ContentTypes.ApplicationJson, ContentTypes.TextPlain)]
```

Comment:
I should remove GetCapturedParameters and rework CaptureParameters to write the captured parameters to the specified egress provider